src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -0,0 +1,99 @@
+    /// </summary>
+    public class TelemetryActivator : ITelemetryActivator
+    {
+        private TelemetryClient telemetryClient;
+        private IOptions<DurableTaskOptions> options;
+
+        /// <summary>
+        /// Constructor for activating Distributed Tracing.
+        /// </summary>
+        /// <param name="options">DurableTask options.</param>
+        public TelemetryActivator(IOptions<DurableTaskOptions> options)
+        {
+            this.options = options;
+        }
+
+        /// <summary>
+        /// OnSend is an action that enable to hook of sending telemetry.
+        /// You can use this property for testing.
+        /// </summary>
+        public Action<ITelemetry> OnSend { get; set; } = null;
+
+        /// <summary>
+        /// Initialize is initialize the telemetry client.
+        /// </summary>
+        public void Initialize()
+        {
+            this.SetUpDistributedTracing();
+
+            this.SetUpTelemetryClient();
+            this.SetUpTelemetryCallbacks();
+        }
+
+        private void SetUpDistributedTracing()
+        {
+            DurableTaskOptions durableTaskOptions = this.options.Value;
+            CorrelationSettings.Current.EnableDistributedTracing =
+                durableTaskOptions.Tracing.DistributedTracingEnabled;
+            CorrelationSettings.Current.Protocol =
+                durableTaskOptions.Tracing.DistributedTracingProtocol == Protocol.W3CTraceContext.ToString()
+                    ? Protocol.W3CTraceContext
+                    : Protocol.HttpCorrelationProtocol;
+        }
+
+        private void SetUpTelemetryCallbacks()
+        {
+            CorrelationTraceClient.SetUp(
+                (TraceContextBase requestTraceContext) =>
+                {
+                    requestTraceContext.Stop();
+
+                    var requestTelemetry = requestTraceContext.CreateRequestTelemetry();
+                    this.telemetryClient.TrackRequest(requestTelemetry);
+                },
+                (TraceContextBase dependencyTraceContext) =>
+                {
+                    dependencyTraceContext.Stop();
+                    var dependencyTelemetry = dependencyTraceContext.CreateDependencyTelemetry();
+                    this.telemetryClient.TrackDependency(dependencyTelemetry);
+                },
+                (Exception e) =>
+                {
+                    this.telemetryClient.TrackException(e);
+                });
+        }
+
+        private void SetUpTelemetryClient()
+        {
+            TelemetryConfiguration config = TelemetryConfiguration.CreateDefault();
+            if (this.OnSend != null)
+            {
+                config.TelemetryChannel = new NoOpTelemetryChannel { OnSend = this.OnSend };
+            }
+
+            var telemetryInitializer = new DurableTaskCorrelationTelemetryInitializer();
+
+            telemetryInitializer.ExcludeComponentCorrelationHttpHeadersOnDomains.Add("127.0.0.1");
```

Comment:
Why do we need this?

Reply:
@TsuyoshiUshio , do you know?

Reply:
As I remember correctly, It is for reducing local debugging noize.  Dependency telemetry is automatically tracked by the Application Insights. That means, it automatically tracked the storage queue/table call via http. I tried to avoid it. If there is a good way, I don't matter.

Reply:
Added to ToDos issues